internal/controller/clusteraddon_controller.go

```diff
@@ -1141,37 +1168,78 @@ func getDynamicResourceAndEvaluateCEL(ctx context.Context, dynamicClient *dynami
 			gvr.Version = splittedAPIVersion[0]
 		}
 
-		mapper := restmapper.NewDeferredDiscoveryRESTMapper(memory.NewMemCacheClient(discoveryClient))
+		mapper := restmapper.NewDeferredDiscoveryRESTMapper(memory.NewMemCacheClient(in.discoverClient))
 		mapping, err := mapper.RESTMapping(schema.GroupKind{Group: gvr.Group, Kind: gvr.Kind}, gvr.Version)
 		if err != nil {
-			return fmt.Errorf("error creating rest mapping: %w", err)
+			conditions.MarkFalse(
+				in.clusterAddon,
+				csov1alpha1.EvaluatedCELCondition,
+				csov1alpha1.CRDNotExistingReason,
+				clusterv1.ConditionSeverityError,
+				"error creating rest mapping for CEL expression: %s", err.Error(),
+			)
+
+			return false, false, fmt.Errorf("error creating rest mapping for CEL expression: %w", err)
 		}
+		conditions.Delete(in.clusterAddon, csov1alpha1.EvaluatedCELCondition)
 
-		resource := dynamicClient.Resource(mapping.Resource)
+		resource := in.dynamicClient.Resource(mapping.Resource)
 
 		unstructuredObj, err := resource.Namespace(object.Namespace).Get(ctx, object.Name, metav1.GetOptions{})
 		if err != nil {
-			return fmt.Errorf("failed to get object(name: %s, namespace: %s) from dynamic client: %w", object.Name, object.Namespace, err)
+			if !apierrors.IsNotFound(err) {
+				return false, false, fmt.Errorf("failed to get object(name: %s, namespace: %s) from dynamic client: %w", object.Name, object.Namespace, err)
+			}
+
+			conditions.MarkFalse(
+				in.clusterAddon,
+				csov1alpha1.EvaluatedCELCondition,
+				csov1alpha1.ObjectNotFoundReason,
+				clusterv1.ConditionSeverityError,
+				"object(%s/%s) from dynamic client for CEL expression is not found: %s", object.Namespace, object.Name, err.Error(),
+			)
+
+			return false, false, fmt.Errorf("failed to get object(%s/%s) from dynamic client for CEL expression: %w", object.Namespace, object.Name, err)
```

Comment:
@janiskemper is there a reason why both strings are different?

Suggestion:

```go
err2 := fmt.fmt.Errorf("failed to get object(%s/%s) from dynamic client for CEL expression: %w", object.Namespace, object.Name, err)

conditions.MarkFalse(..., err2.Error())

return false, false, err2
```

Just a comment, not merge blocking.